ios/brave-ios/Sources/BraveWallet/WalletUserAssetManager.swift

```diff
@@ -17,60 +17,72 @@ public protocol WalletUserAssetDataObserver: AnyObject {
 }
 
 public protocol WalletUserAssetManagerType: AnyObject {
-  /// Return all visible or all invisible user assets in form of `NetworkAssets`
-  func getAllUserAssetsInNetworkAssetsByVisibility(
+  /// Return all visible or invisible user assets from `WalletService` that are on given a list of networks in form of `NetworkAssets`
+  /// The result will exclude any user assets that are previously marked as deleted by user
+  func getUserAssets(
     networks: [BraveWallet.NetworkInfo],
     visible: Bool
-  ) -> [NetworkAssets]
-  /// Return all user assets in form of `NetworkAssets`
+  ) async -> [NetworkAssets]
+  /// Return all user assets from `WalletService` that are on given a list of networks in form of `NetworkAssets`,
+  /// given the option to include or exclude any user assets that are marked as deleted by user
   func getAllUserAssetsInNetworkAssets(
     networks: [BraveWallet.NetworkInfo],
     includingUserDeleted: Bool
-  ) -> [NetworkAssets]
-  /// Return all spam or non-spam user assets in form of `NetworkAssets`
-  func getAllUserNFTs(networks: [BraveWallet.NetworkInfo], isSpam: Bool) -> [NetworkAssets]
+  ) async -> [NetworkAssets]
+  /// Return all spam or non-spam user assets (NFTs) from `WalletService` that are on given a list of networks in form of `NetworkAssets`
+  /// The result will exclude any user assets that are previously marked as deleted by user
+  func getAllUserNFTs(
+    networks: [BraveWallet.NetworkInfo],
+    isSpam: Bool
+  ) async -> [NetworkAssets]
   /// Return all user marked deleted user assets
-  func getAllUserDeletedNFTs() -> [WalletUserAsset]
+  func getAllUserDeletedUserAssets() -> [WalletUserAsset]
   /// Return a `WalletUserAsset` with a given `BraveWallet.BlockchainToken`
   func getUserAsset(_ asset: BraveWallet.BlockchainToken) -> WalletUserAsset?
-  /// Add a `WalletUserAsset` representation of the given
-  /// `BraveWallet.BlockchainToken` to CoreData
-  func addUserAsset(_ asset: BraveWallet.BlockchainToken) async
+  /// Add asset to `WalletService` as well as adding a
+  /// `WalletUserAsset` representation in CoreData
+  /// This should be only used when user manually adding a custom asset.
+  func addUserAsset(
+    _ token: BraveWallet.BlockchainToken,
+    isAutoDiscovery: Bool
+  ) async -> Bool
```

Comment:
```suggestion
  @discardableResult func addUserAsset(
    _ token: BraveWallet.BlockchainToken,
    isAutoDiscovery: Bool
  ) async -> Bool
```

Causes warning in `CryptoStore` as we are not using the result